gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -532,6 +538,73 @@ private Map<String, AttributeMetadata> remapAttributeMetadataByResultName(
         attributeMetadataByIdMap);
   }
 
+  private org.hypertrace.gateway.service.v1.common.Filter createEntityIdAttributeFilter(
+      List<String> entityIdAttributes,
+      org.hypertrace.gateway.service.v1.common.Operator operator,
+      List<String> entityIds) {
+    if (entityIdAttributes.size() != 1) {
+      throw new RuntimeException("Cannot have more than one id attribute for an entity");
+    }
+
+    if (entityIds.isEmpty()) {
+      // Having empty entity ids is valid filter because this means that
```

Comment:
It's fine as a quick fix for the issue, but this is not a great long term solution to fire a query we know will return empty. Can we leave a TODO to come up with a better solution here?